### DISCUSSION TITLE
is_manifold vs is_closed

### DISCUSSION THREAD

**Discussion by danafd** [marcomusy/vedo#993] [General]
Hello, 
I'm fairly new to analysing meshes and sometimes get confused with terminologies that I am not familiar with. What is the difference between the is_manifold() and is_closed() functions for meshes? And which one do I need to be true to measure the true volume of a mesh ( using the volume() function)?

**Comment by marcomusy** [comment]
Hi, for a mesh to be manifold it must not have no edge with more than 2 faces attached to it. 
Try
```
>>> from vedo import *
>>> Circle().is_manifold()
True
>>> Circle().is_closed()
False
>>> Sphere().is_closed()
True
>>> Sphere().is_manifold()
True
```

**Reply by danafd** [reply]
Thanks for the clarification @marcomusy!
I have the following 3D shape (binary segmentation from microscopy image of embryonic forebrain). I would like to quantify its volume but I'm wary of the possibility that the volume() function would give false results because the mesh is not closed. I show the image at an angle where it is clear that the 2 big lobes are not connected in space. 
<img width="619" alt="Screenshot 2023-12-20 at 15 14 44" src="https://github.com/marcomusy/vedo/assets/49127216/0fdf84b1-62b3-4e19-bdba-5d0a286c4505">

`msh.is_manfold()` returns `True`
`msh.is_closed()` returns `False`

Does the `msh.volume()` work here? 

Thanks for your help!

**Reply by marcomusy** [reply]
Forn non closed meh the result may be not precise.
I would check where it is not closed with `mesh.boundaries()`.
you can also use `mesh.split()` to split in 2 (or more pieces) and evaluate the volumes separately.
you can also use `.extract_largest_region()` to remove the noise.

**Reply by danafd** [reply]
Thanks for the tips!
